frontend/Components/Header.tsx

```diff
@@ -2,23 +2,18 @@ import * as React from 'react';
 import {
 	Link,
 } from 'react-router-dom';
-export interface HeaderParams{
-	pageName: string;
-}
-export class MainHeader extends React.Component<HeaderParams> {
-	private readonly pageName: string;
 
-	constructor(params: HeaderParams) {
-		super(params);
-		this.pageName = params.pageName;
-	}
+export class Header extends React.Component {
```

Comment:
If you take a look at the definitions for `React.Component` you'll notice it has two generics, `P` and `S`:

```
class Component<P, S> {
```

`P` is for declaring the type of properties, and `S` is for declaring the type of the state. What I am implying here is you should add an interface for the state and pass it into the generic so that Typescript can do its job.